cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -184,39 +196,38 @@ def update_barcode_mismatch_values_for_sample(
             sample_to_update.barcode_mismatches_2 = 0
 
 
-def adapt_indexes_for_sample(
+def pad_and_reverse_complement_sample_indexes(
     sample: FlowCellSample, index_cycles: int, reverse_complement: bool
 ) -> None:
     """Adapts the indexes of sample.
     1. Pad indexes if needed so that all indexes have a length equal to the number of index reads
     2. Takes the reverse complement of index 2 in case of the new NovaSeq software control version
     (1.7) in combination with the new reagent kit (version 1.5).
-    3. Assigns the indexes to the sample attributes index1 and index2."""
+    3. Assigns the indexes to the sample attributes index and index2."""
     index1, index2 = get_index_pair(sample=sample)
     index_length = len(index1)
-    if is_padding_needed(index_cycles=index_cycles, sample_index_length=index_length):
+    if isinstance(sample, FlowCellSampleBcl2Fastq) and is_padding_needed(
+        index_cycles=index_cycles, sample_index_length=index_length
+    ):
         LOG.debug("Padding indexes")
         index1 = pad_index_one(index_string=index1)
         index2 = pad_index_two(index_string=index2, reverse_complement=reverse_complement)
+    LOG.debug(f"Padding not necessary for sample {sample.sample_id}")
     if reverse_complement:
         index2 = get_reverse_complement_dna_seq(index2)
     sample.index = index1
     sample.index2 = index2
 
 
-def adapt_samples(
-    samples: List[FlowCellSample],
+def update_indexes_for_samples(
+    samples: List[Union[FlowCellSampleBCLConvert, FlowCellSampleBcl2Fastq]],
     run_parameters: RunParameters,
     reverse_complement: bool,
 ) -> None:
-    """Adapt the indexes and updates the barcode mismatch values of the samples."""
+    """Updates the values to the fields index1 and index 2 of samples."""
     index_cycles: int = run_parameters.index_length
     for sample in samples:
-        if run_parameters.sequencer == Sequencers.NOVASEQX:
-            update_barcode_mismatch_values_for_sample(
-                sample_to_update=sample, samples_to_compare_to=samples
-            )
-        adapt_indexes_for_sample(
+        pad_and_reverse_complement_sample_indexes(
             sample=sample,
```

Comment:
Is this function not just obsolete? Why not just use pad_and_reverse_complemet_sample_indexes and put the for loop in there?

Reply:
for SRP and clarity, I thought the inner function should do the update for one sample and the outer function loop over all functions. This way the functions are easier to test. Do you have any reason why we wanted it as one big function? I think `pad_and_reverse_complement_sample_indexes` is already a bit complex on its own

Reply:
done